plugins/arElasticSearchPlugin/lib/arElasticSearchPlugin.class.php

```diff
@@ -365,6 +367,30 @@ public function partialUpdate($object, $data)
         }
     }
 
+    /**
+     * Update all fields specified in script for matching terms in one request.
+     * Use for bulk update e.g. update publication status by id.
+     *
+     * @param array $terms
+     * @param array $script
+     * @param mixed $options
+     */
+    public function updateByQuery($terms, $script, $options = [])
+    {
+        $queryBool = new \Elastica\Query\BoolQuery();
+
+        foreach ($terms as $k => $v) {
+            $query = new \Elastica\Query\Term();
+            $query->setTerm($k, $v);
+            $queryBool->addShould($query);
+        }
+
+        $queryScript = \Elastica\Script\AbstractScript::create($script);
+        $response = $this->index->updateByQuery($queryBool, $queryScript, $options);
+
+        return $response->getData();
+    }
+
```

Comment:
I'm not sure this wrapper ads a lot and I think it could get easily out of hand (allowing other type of queries, etc.). For now, building the query/script and calling `QubitSearch::getInstance()->index->updateByQuery()` from the job directly could be enough.

Reply:
If you use partial update for the resource update I think we still don't need this abstraction. Even if we keep it, we don't need the bool query yet.

Reply:
I have removed the wrapper and removed the bool query.